corehq/util/quickcache.py

```diff
@@ -1,329 +1,26 @@
-# coding=utf-8
-from collections import namedtuple
-import functools
-import hashlib
-import inspect
-from inspect import isfunction
-import logging
-from django.core.cache import caches as django_caches
-from corehq.util.soft_assert.api import soft_assert
+from __future__ import absolute_import
+import warnings
+from quickcache.django_quickcache import get_django_quickcache
 
-logger = logging.getLogger('quickcache')
 
+from corehq.util.soft_assert import soft_assert
 
-class CacheWithTimeout(namedtuple('CacheWithTimeout', ['cache', 'timeout'])):
+quickcache_soft_assert = soft_assert(
+    notify_admins=True,
+    fail_if_debug=False,
+    skip_frames=5,
+)
 
-    def get(self, key, default=None):
-        return self.cache.get(key, default=default)
+quickcache = get_django_quickcache(timeout=5 * 60, memoize_timeout=10,
+                                   assert_function=quickcache_soft_assert)
 
-    def set(self, key, value):
-        return self.cache.set(key, value, timeout=self.timeout)
 
-    def delete(self, key):
-        return self.cache.delete(key)
+def skippable_quickcache(*args, **kwargs):
+    warnings.warn(
+        "skippable_quickcache is deprecated. Use quickcache with skip_arg instead.",
+        DeprecationWarning
+    )
+    return quickcache(*args, **kwargs)
```

Comment:
This is just in case some open PRs are using this / for the transition. After steady state, we can just remove this. (Could probably remove it now, but was easy to be conservative and keep this shim.)